src/Widgets/DwellClickTimer.vala

```diff
@@ -0,0 +1,139 @@
+//
+//  Copyright 2020 elementary, Inc. (https://elementary.io)
+//
+//  This program is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+namespace Gala {
+    public class DwellClickTimer : Clutter.Actor, Clutter.Animatable {
+        private const int WIDTH_PX = 60;
+        private const int HEIGHT_PX = 60;
+        private const string BACKGROUND_COLOR = "#64baff";
+        private const double BACKGROUND_OPACITY = 0.7;
+        private const uint BORDER_WIDTH_PX = 3;
+
+        private const double START_ANGLE = 3 * Math.PI / 2;
+
+        private int scaling_factor = 1;
+
+        private Cogl.Pipeline pipeline;
+        private Clutter.PropertyTransition transition;
+        private Cairo.Pattern stroke_color;
+        private Cairo.Pattern fill_color;
+
+        public weak WindowManager wm { get; construct; }
+
+        public double angle { get; set; }
+
+        public DwellClickTimer (WindowManager wm) {
+            Object (wm: wm);
+        }
+
+        construct {
+            visible = false;
+            reactive = false;
+
+            pipeline = new Cogl.Pipeline (Clutter.get_default_backend ().get_cogl_context ());
+
+            transition = new Clutter.PropertyTransition ("angle");
+            transition.set_progress_mode (Clutter.AnimationMode.EASE_OUT_QUAD);
+            transition.set_animatable (this);
+            transition.set_from_value (START_ANGLE);
+            transition.set_to_value (START_ANGLE + (2 * Math.PI));
+
+            transition.new_frame.connect (() => {
+                queue_redraw ();
+            });
+
+            var rgba = Gdk.RGBA ();
+            rgba.parse (BACKGROUND_COLOR);
```

Comment:
As for now, it's better to hardcode the RGB values of the color and feed set them directly into the struct, rather than having the class parse the same string everytime on construction.

Reply:
This class is only constructed once when Gala starts. I felt it was more beneficial to have a string constant at the top that was easily copied and pasted from a palette etc than it was to need to change 3 numbers somewhere in the file when you need to change the colour. Happy to change it if you feel it's necessary though.

Reply:
I think since we'll want to get the color from the stylesheet anyways that this is probably okay for now